petab/core.py

```diff
@@ -4,63 +4,145 @@
+        )
+
     def _load_sbml(self):
-        """Load SBML model"""
+        """
+        Load SBML model.
+        """
 
-        # sbml_reader and sbml_document must be kept alive. Otherwise operations on sbml_model will segfault
+        # sbml_reader and sbml_document must be kept alive.
+        # Otherwise operations on sbml_model will segfault
         self.sbml_reader = libsbml.SBMLReader()
         self.sbml_document = self.sbml_reader.readSBML(self.sbml_file)
         self.sbml_model = self.sbml_document.getModel()
 
     def get_constant_parameters(self):
-        """Provide list of IDs of parameters which are fixed (i.e. not subject to optimization, no sensitivities w.r.t. these parameters are required)"""
-
-        return list(set(self.condition_df.columns.values.tolist()) - {'conditionId', 'conditionName'})
+        """
+        Provide list of IDs of parameters which are fixed (i.e. not subject
+        to optimization, no sensitivities w.r.t. these parameters are
+        required).
+        """
+        columns_set = set(self.condition_df.columns.values)
+        return list(columns_set - {'conditionId', 'conditionName'})
+    
+    def get_optimization_parameters(self):
+        """
+        Return list of optimization parameter ids.
+        """
+        return get_optimization_parameters(self.parameter_df)
 
-    def get_observables(self):
-        """Returns dictionary of observables definitions
+    def get_dynamic_simulation_parameters(self):
+        return get_dynamic_simulation_parameters(self.sbml_model, self.parameter_df)
 
-        see `assignment_rules_to_dict` for details"""
+    def get_dynamic_parameters_from_sbml(self):
+        """
+        Provide list of IDS of parameters which are dynamic, i.e. not
+        fixed.
+        See get_dynamic_parameters_from_sbml.
+        """
+        return get_dynamic_parameters_from_sbml(self.sbml_model)
 
-        return get_observables(self.sbml_model)
+    def get_observables(self, remove=False):
+        """
+        Returns dictionary of observables definitions
+        See `assignment_rules_to_dict` for details.
+        """
 
-    def get_sbml_sigmas(self):
-        """Return dictionary of observableId => sigma as defined in the SBML model
+        return get_observables(sbml_model=self.sbml_model, remove=remove)
 
-        This does not include parameter mappings defined in the measurement table.
+    def get_sigmas(self, remove=False):
+        """
+        Return dictionary of observableId => sigma as defined in the SBML
+        model.
+        This does not include parameter mappings defined in the measurement
+        table.
         """
 
-        return get_sigmas(sbml_model=self.sbml_model)
+        return get_sigmas(sbml_model=self.sbml_model, remove=remove)
+
+    def get_simulation_conditions_from_measurement_df(self):
+        return get_simulation_conditions_from_measurement_df(self.measurement_df)
+
+    def map_par_sim_to_par_opt(self):
```

Comment:
Having both `map_par_sim_to_par_opt` and `get_simulation_to_optimization_parameter_mapping` is a bit confusing.

Reply:
Can you add test cases, or change the old ones to use this function?

Reply:
Done.